### PR TITLE
Remove deprecated annotations-based L7 visibility

### PR DESCRIPTION
**Pull request by tklauser**
Previously it was possible to enable L7 protocol visibility using the `policy.cilium.io/proxy-visibility` Pod annotation. That method has various bugs, is no longer supported and has been deprecated since Cilium 1.15, see https://github.com/cilium/cilium/pull/28449.

See commits for details.

### CODE REVIEW

**Review by derailed — APPROVED** [review]
@tklauser Nice clean up!

**Review by YutaroHayakawa — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

**Review by rolinh — APPROVED** [review]
Nice cleanup!

File: Documentation/observability/visibility.rst [hunk 1]
```diff
@@ -19,16 +19,10 @@ will only provide visibility into L3/L4 packet events. If you want L7
 protocol visibility, you can use L7 Cilium Network Policies (see :ref:`l7_policy`).
 
 
-.. note::
```
Comment (squeed) [inline Documentation/observability/visibility.rst:22]: We should explicitly note this removal in the upgrading notes.

**Review by squeed — APPROVED** [review]
Looks good. Would like to have @jrajahalme's review before merging.

There's a minor docs change needed, otherwise this is good.

File: Documentation/observability/visibility.rst (hunk 1 above)
Comment (tklauser) [reply Documentation/observability/visibility.rst:22]: This PR already does that in the upgrade notes here: https://github.com/cilium/cilium/pull/35019/files#diff-76c7d3d61bd3c6afeb8e936896c7a01046d530840abf499fd118e88e75d1a504R314

Do we need to mention this somewhere else or should the note be more verbose?

**Comment by tklauser** [comment]
@jrajahalme @learnitall could you please take a look?

**Review by jrajahalme — APPROVED** [review]
Looking great, thanks for the clean-up!

**Review by aanm — APPROVED** [review]
Could we backport at least the test changes up to v1.15 branch?

**Comment by tklauser** [comment]
> Could we backport at least the test changes up to v1.15 branch?

Sure, will open separate backports for these.

**Review by qmonnet — APPROVED** [review]

**Comment by tklauser** [comment]
> > Could we backport at least the test changes up to v1.15 branch?
> 
> Sure, will open separate backports for these.

`v1.16`: https://github.com/cilium/cilium/pull/35151
`v1.15`: https://github.com/cilium/cilium/pull/35152

**Pull request merged**